packages/ckb/src/utils/cell-dep.ts

```diff
@@ -0,0 +1,107 @@
+import axios from 'axios';
+import { getBtcTimeLockDep, getRgbppLockDep, getUniqueTypeDep, getXudtDep } from '../constants';
+
+interface CellDepsObject {
+  rgbpp: {
+    mainnet: CKBComponents.CellDep;
+    testnet: CKBComponents.CellDep;
+  };
+  btcTime: {
+    mainnet: CKBComponents.CellDep;
+    testnet: CKBComponents.CellDep;
+  };
+  xudt: {
+    testnet: CKBComponents.CellDep;
+  };
+  unique: {
+    testnet: CKBComponents.CellDep;
+  };
+}
+const GITHUB_CELL_DEPS_JSON_URL =
+  'https://raw.githubusercontent.com/ckb-cell/typeid-contract-cell-deps/main/deployment/cell-deps.json';
+
+const fetchCellDepsJson = async () => {
+  try {
+    const response = await axios.get(GITHUB_CELL_DEPS_JSON_URL);
+    return response.data as CellDepsObject;
+  } catch (error) {
+    // console.error('Error fetching cell deps:', error);
+  }
```

Comment:
```suggestion
const fetchCellDepsJson = async () => {
  try {
    const response = await axios.get(GITHUB_CELL_DEPS_JSON_URL, { timeout: 2000 });
    return response.data as CellDepsObject;
  } catch (error) {
    console.error('Error fetching cell deps:', error);
  }
```
### Suggestion
Abort the request when the request takes longer than `timeout`.
The `timeout` value can be tuned after some testing.

Reply:
I agree to set the timeout for the fetching and it will reduce the waiting time for users.

But I don't think showing the fetching error for developers is a good idea.

If the fetching error occurs, the `cellDeps` from `constants` will be used and it will work well, so developers don't need to know the fetching error.

I think the error logs will be only used to debug for ourselves.

Reply:
LGTM

Reply:
> I think the error logs will be only used to debug for ourselves.

LGTM

Reply:
https://github.com/ckb-cell/rgbpp-sdk/pull/191#issuecomment-2134782766